src/cryptography/hazmat/primitives/asymmetric/dh.py

```diff
@@ -87,9 +87,6 @@ def __init__(self, p, g, q=None):
         if q is not None and not isinstance(q, six.integer_types):
             raise TypeError("q must be integer or None")
 
-        if q is None and g not in (2, 5):
-            raise ValueError("DH generator must be 2 or 5")
```

Comment:
I'm mostly trying to reconcile the existence of `DH_check` with the fact that we don't (and can't) validate here, but g=1 would still be a very bad thing to put here, so I think `g > 1` is probably what the modified check should be if we're going to merge this.